### PR TITLE
cpu/xive.py: Fixed test_storeEOI for SLES15.x

### PR DESCRIPTION
**Pull request by rohan-r-deshpande**
- Added fix for test_storeEOI.
- This fix handles a scenario wherein the store-eoi flag is identified either by using the ipis file or the interrupts file, whichever available.
- These files are checked under the following path: "/sys/kernel/debug/powerpc/xive/"

Signed-off-by: Rohan Deshpande <rohan_d@linux.ibm.com>

### CODE REVIEW

**Comment by rohan-r-deshpande** [comment]
**[RHEL 8.7]**

```
[root@ltcden9-lp1 cpu]# ls -l /sys/kernel/debug/powerpc/xive 
-r--------. 1 root root 0 Nov 28 19:29 /sys/kernel/debug/powerpc/xive


[root@ltcden9-lp1 cpu]# avocado run --test-runner runner xive.py
JOB ID     : 36cadb8742ab128315eb5c81c2462461e9936cdb
JOB LOG    : /home/rohan/avocado-fvt-wrapper/results/job-2022-12-05T18.58-36cadb8/job.log
 (1/4) xive.py:XIVE.test_intr_mode: PASS (0.52 s)
 (2/4) xive.py:XIVE.test_storeEOI: PASS (0.46 s)
 (3/4) xive.py:XIVE.test_verify_xive_cmdline: PASS (0.44 s)
 (4/4) xive.py:XIVE.test_verify_xive_config: PASS (0.47 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/rohan/avocado-fvt-wrapper/results/job-2022-12-05T18.58-36cadb8/results.html
JOB TIME   : 2.47 s
[root@ltcden9-lp1 cpu]#
```

**[SLES15-SP4]**

```
localhost:~ # ls -l /sys/kernel/debug/powerpc/
total 0
. . . .
. . . . 
drwxr-xr-x 3 root root 0 Dec  5 12:18 xive
. . . .


localhost:~ # ls -l /sys/kernel/debug/powerpc/xive/
total 0
drwxr-xr-x 2 root root 0 Dec  5 12:18 eqs
-r-------- 1 root root 0 Dec  5 12:18 interrupts
-r-------- 1 root root 0 Dec  5 12:18 ipis
localhost:~ # 


localhost: # avocado run --test-runner runner xive.py
JOB ID     : 015afeb583a2011528c1c607c5a79eeda2e51fc3
JOB LOG    : /home/rohan/avocado-fvt-wrapper/results/job-2022-12-05T18.34-015afeb/job.log
 (1/4) xive.py:XIVE.test_intr_mode: PASS (0.33 s)
 (2/4) xive.py:XIVE.test_storeEOI: PASS (0.33 s)
 (3/4) xive.py:XIVE.test_verify_xive_cmdline: PASS (0.33 s)
 (4/4) xive.py:XIVE.test_verify_xive_config: PASS (0.32 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/rohan/avocado-fvt-wrapper/results/job-2022-12-05T18.34-015afeb/results.html
JOB TIME   : 34.49 s

```

File: cpu/xive.py [hunk 1]
```diff
@@ -52,6 +54,14 @@ def setUp(self):
...
         self.no_config_parameter = []
 
+        self.detected_distro = distro.detect()
+        if self.detected_distro.name not in ['SuSE', 'rhel']:
+            self.cancel('Unsupported OS %s' % self.detected_distro.name)
+        if (self.detected_distro.name == 'SuSE'):
+            self.device = "eth0"
+        elif (self.detected_distro.name == 'rhel'):
+            self.device = "net0"
+
```
Comment (sacsant) [inline cpu/xive.py:64]: This does not seem related to the changes being proposed for XIVE. Can you remove it ?

File: cpu/xive.py [hunk 2]
```diff
@@ -72,15 +82,34 @@ def test_storeEOI(self):
...
+                    # store-eoi value can be enabled or disable via kernel
+                    # command line or using sysfs debug directory.
+                    match = re.search("Y", flags) or re.search("N", flags)
+                else:
+                    if (self.detected_distro.name == 'rhel'):
+                        self.cancel("Test Skipped! Invalid scenario for RHEL")
+                    elif (self.detected_distro.name == 'SuSE'):
+                        interrupts_file = os.path.join(xive_path, "interrupts")
+                        ipis_file = os.path.join(xive_path, "ipis")
+
```
Comment (sacsant) [inline cpu/xive.py:99]: We don't necessarily need to add checks for distro releases. We can check for which files are present and take appropriate action. Also it's not required to check for both ipis and interrupts file, only one will suffice.

We can define a new variable xive_file and assign the value based on which file should be used to read store-eoi flag information. We then use this variable and pass it to read_file() .. rest of the code then can remain unchanged

**Review by sacsant — CHANGES_REQUESTED** [review]

File: cpu/xive.py [hunk 3]
```diff
@@ -19,6 +19,8 @@
 
 from avocado import Test
 from avocado.utils import genio
+from avocado.utils import distro
+from avocado.utils import process
```
Comment (Naresh-ibm) [inline cpu/xive.py:23]: I think you are not using the process, so remove this.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: cpu/xive.py (hunk 1 above)
Comment (rohan-r-deshpande) [reply cpu/xive.py:64]: > This does not seem related to the changes being proposed for XIVE. Can you remove it ?

Looks like I checked-in the code changes which were intended for a different scenario that I was working on, in the same file. Sorry, will get it removed.

File: cpu/xive.py (hunk 2 above)
Comment (rohan-r-deshpande) [reply cpu/xive.py:99]: Okay, noted. On it!

> We don't necessarily need to add checks for distro releases. We can check for which files are present and take appropriate action. Also it's not required to check for both ipis and interrupts file, only one will suffice.
> 
> We can define a new variable xive_file and assign the value based on which file should be used to read store-eoi flag information. We then use this variable and pass it to read_file() .. rest of the code then can remain unchanged

File: cpu/xive.py (hunk 1 above)
Comment (rohan-r-deshpande) [reply cpu/xive.py]: > > This does not seem related to the changes being proposed for XIVE. Can you remove it ?
> 
> Looks like I checked-in the code changes which were intended for a different scenario that I was working on, in the same file. Sorry, will get it removed.

Done.

File: cpu/xive.py (hunk 2 above)
Comment (rohan-r-deshpande) [reply cpu/xive.py]: > Okay, noted. On it!
> 
> > We don't necessarily need to add checks for distro releases. We can check for which files are present and take appropriate action. Also it's not required to check for both ipis and interrupts file, only one will suffice.
> > We can define a new variable xive_file and assign the value based on which file should be used to read store-eoi flag information. We then use this variable and pass it to read_file() .. rest of the code then can remain unchanged

@sacsant Done.

File: cpu/xive.py (hunk 3 above)
Comment (rohan-r-deshpande) [reply cpu/xive.py]: Sure, done!

> I think you are not using the process, so remove this.

File: cpu/xive.py [hunk 4]
```diff
@@ -72,15 +72,26 @@ def test_storeEOI(self):
...
-
-            flags = genio.read_file(xive_path)
-            if is_dir:
-                # store-eoi value can be enabled or disable via kernel command
-                # line or using sysfs debug directory.
-                match = re.search("Y", flags) or re.search("N", flags)
+                # store_eoi_file = os.path.join(xive_path, "store-eoi")
+                xive_file = os.path.join(xive_path, "store-eoi")
+                is_file_exists = os.path.exists(xive_file)
+                if is_file_exists:
```
Comment (abdhaleegit) [inline cpu/xive.py:78]: if os.path.exists(xive_file):   will eleminate one line and one assigment

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

**Comment by abdhaleegit** [comment]
@rohan-r-deshpande a common convention we follow on commit message is to have one empty line between commit header line and the actual commit description sample

```
Fixing this error in the test.py

This commit fixes a problem in this way and in this files
This commit fixes a problem in this way and in this files

Sign-off ...
```

File: cpu/xive.py (hunk 4 above)
Comment (rohan-r-deshpande) [reply cpu/xive.py]: > if os.path.exists(xive_file): will eleminate one line and one assigment

@abdhaleegit Done. Please check. Thanks!

**Comment by rohan-r-deshpande** [comment]
> @rohan-r-deshpande a common convention we follow on commit message is to have one empty line between commit header line and the actual commit description sample
> 
> ```
> Fixing this error in the test.py
> 
> This commit fixes a problem in this way and in this files
> This commit fixes a problem in this way and in this files
> 
> Sign-off ...
> ```

Done!

**Review by abdhaleegit — APPROVED** [review]
LGTM

File: cpu/xive.py [hunk 5]
```diff
@@ -72,15 +72,25 @@ def test_storeEOI(self):
...
             if is_dir:
                 # If xive is a directory then read from store-eoi file
-                xive_path = "/sys/kernel/debug/powerpc/xive/store-eoi"
-
-            flags = genio.read_file(xive_path)
-            if is_dir:
-                # store-eoi value can be enabled or disable via kernel command
-                # line or using sysfs debug directory.
-                match = re.search("Y", flags) or re.search("N", flags)
+                # store_eoi_file = os.path.join(xive_path, "store-eoi")
```
Comment (Naresh-ibm) [inline cpu/xive.py:75]: @rohan-r-deshpande is this line required?

Comment (PraveenPenguin) [reply cpu/xive.py:75]: @rohan-r-deshpande can you please address this review comment

**Review by abdhaleegit — APPROVED** [review]
LGTM

File: cpu/xive.py (hunk 5 above)
Comment (rohan-r-deshpande) [reply cpu/xive.py]: @Naresh-ibm @PraveenPenguin Done. Please check!

**Review by PraveenPenguin — APPROVED** [review]
LGTM
